File: day01/day01-part1.php
<?php

$input = file(__DIR__ . '/real-input');

$start = 50;
$zeroCount = 0;

foreach($input as $instruction) {
    $dir = $instruction[0];
    $amt = (int) substr($instruction, 1);

    $start += $dir === 'L'
        ? -$amt
        : $amt;
    
    // wrap to between 0 and 99
    $start = ($start + 100) % 100;

    // part 1: if we land on exactly zero, count it
    if ($start === 0) $zeroCount++;
}

printf("Number of times zero was hit: %d\n", $zeroCount);

File: day01/day01-part1.py

def readFile():
    with open("python-input") as file:
        return [line.strip() for line in file]

contents = readFile()

pos = 50
ans = 0

for line in contents:
    dir, amt = line[0], int(line[1:])

    if dir == "L":
        pos -= amt
    else:
        pos += amt

    pos = (pos + 100) % 100

    # If we land exactly on zero, count towards answer
    if pos == 0:
        ans += 1

print(ans)
